src/homepageExperience/components/steps/arduino/InstallDependencies.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  ComponentColor,
+  Button,
+  ButtonGroup,
+  Orientation,
+} from '@influxdata/clockface'
+
+import React, {FC, useEffect, useState} from 'react'
+import {useDispatch} from 'react-redux'
+import {getBuckets} from 'src/buckets/actions/thunks'
+import {event} from 'src/cloud/utils/reporting'
+import {keyboardCopyTriggered, userSelection} from 'src/utils/crossPlatform'
+
+export const InstallDependencies: FC = () => {
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(getBuckets())
+  }, [])
+
+  const headingWithMargin = {marginTop: '48px', marginBottom: '0px'}
+
+  const logCopyCodeSnippetMac = () => {
```

Comment:
i think instead of having three functions that are essentially duplicates of one another, we can reduce them into one function that takes OS as a prop and updates the event string based on the passed prop. [Here](https://github.com/influxdata/ui/pull/5275/files#diff-010131b4d0e1a1d9fd41db832a0f323cc3abfabee55a30cb90f8e47fb0ac22fdR26) is what I mean.